src/main/java/org/galatea/starter/entrypoint/SettlementRestController.java

```diff
@@ -34,34 +37,60 @@
 @RestController
 public class SettlementRestController {
 
-  @NonNull
-  SettlementService settlementService;
-
-  public static final String SETTLE_MISSION_PATH = "/settlementEngine";
-  public static final String GET_MISSION_PATH = SETTLE_MISSION_PATH + "/mission/";
-
-  @PostMapping(value = SETTLE_MISSION_PATH, consumes = {MediaType.APPLICATION_JSON_VALUE})
-  public ResponseEntity<Set<String>> settleAgreement(
-      @RequestBody final List<TradeAgreement> agreements) {
-    Set<Long> missionIds = settlementService.spawnMissions(agreements);
-    Set<String> missionIdUris =
-        missionIds.stream().map(id -> GET_MISSION_PATH + id).collect(Collectors.toSet());
-
-    return ResponseEntity.accepted().body(missionIdUris);
-  }
-
-  @GetMapping(value = GET_MISSION_PATH + "{id}", produces = {MediaType.APPLICATION_JSON_VALUE})
-  public ResponseEntity<SettlementMission> getMission(@PathVariable final Long id) {
-
-    Optional<SettlementMission> msn = settlementService.findMission(id);
-
-    if (msn.isPresent()) {
-      return ResponseEntity.ok(msn.get());
-    }
-
-    return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-  }
-
-
+	@NonNull
+	@Autowired
+	SettlementService settlementService;
+
+	public static final String SETTLE_MISSION_PATH = "/settlementEngine";
+	public static final String GET_MISSION_PATH = SETTLE_MISSION_PATH + "/mission/";
+
+	// @PostMapping to link http POST requests to this method
+	// @RequestBody to have the post request body deserialized into a list of TradeAgreement objects
+	@PostMapping(value = SETTLE_MISSION_PATH, consumes = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<RestResponse> settleAgreement(@RequestBody final List<TradeAgreement> agreements,
+			@RequestParam(value = "requestId", required = false) String requestId) {
+
+		// if an external request id was provided, grab it
+		processRequestId(requestId);
+
+		Set<Long> missionIds = settlementService.spawnMissions(agreements);
+		Set<String> missionIdUris = missionIds.stream().map(id -> GET_MISSION_PATH + id).collect(Collectors.toSet());
+
+		// wrap response in RestResponse to get nice Audit details in the response
+		return ResponseEntity.accepted().body(new RestResponse(missionIdUris));
```

Comment:
We should wrap the response in our "RestResponse" in the FuseWebRequestTraceFilter or handle it via an annotation processor so the developer doesn't have to worry about it.    The filter already works with the trace object so it can pull out what it needs from the trace object and stick it on the audit response.